test/OrchardCore.Commerce.Tests.UI/Tests/OrderTests/BehaviorOrderTests.cs

```diff
@@ -0,0 +1,157 @@
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[0].UnitPriceValue"), "10");
+                await context.ClickPublishAsync();
+
+                // Non-existent SKU should result in validation errors being shown and no Product being added.
+                VerifyValidationErrors(
+                    context, "SKU \"NONEXISTENTPRODUCT\" is empty or does not belong to an existing Product."); // #spell-check-ignore-line
+
+                context.Missing(By.Id("OrderPart.LineItems[0].Quantity"));
+                await AddNewProductAsync(context);
+
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[0].Quantity"), "5");
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[0].ProductSku"), "testproductvariant");
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[0].UnitPriceValue"), "10");
+                await context.SetDropdownByTextAsync(By.Name("OrderPart.LineItems[0].UnitPriceCurrencyIsoCode"), "EUR");
+                await context.ClickPublishAsync();
+
+                // Selected currency should be saved.
+                AssertTotal(context, "€", "50");
+                context
+                    .Get(By.Name("OrderPart.LineItems[0].UnitPriceCurrencyIsoCode"))
+                    .GetValue()
+                    .ShouldBe("EUR");
+
+                // For a Product with existing attributes, attributes should be populated and should be selectable.
+                await context.SetDropdownByTextAsync(By.Name("OrderPart.LineItems[0].SelectedAttributes[Size]"), "Medium");
+                await context.ClickPublishAsync();
+
+                context
+                    .Get(By.Name("OrderPart.LineItems[0].SelectedAttributes[Size]"))
+                    .GetValue()
+                    .ShouldBe("Medium");
+
+                await AddNewProductAsync(context);
+
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[1].Quantity"), "5");
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[1].ProductSku"), "testproduct");
+                await context.ClickAndFillInWithRetriesAsync(By.Name("OrderPart.LineItems[1].UnitPriceValue"), "10");
+                await context.ClickPublishAsync();
+
+                // No attributes should be loaded for Product without attributes.
+                context.Missing(By.Name("OrderPart.LineItems[1].SelectedAttributes[Size]"));
+
+                await context.SetDropdownByTextAsync(By.Name("OrderPart.LineItems[1].UnitPriceCurrencyIsoCode"), "CAD");
+                await context.ClickPublishAsync();
+
+                // Selecting non-matching currencies should result in validation errors.
+                VerifyValidationErrors(context, "Selected currencies need to match.");
```

Comment:
I just realized this, but "Selected currencies need to match." should be "Selected currencies must match." instead.